### PR TITLE
users endpoint returning my user data

### PR DESCRIPTION
**Pull request by marcofucci**
This adds a `/users/<my-username>/` endpoint returning my user data.
Access to other's user data is not allowed.

This can be used to get user pk / full name and the prisons
that the user can manages.

### CODE REVIEW

File: mtp_api/apps/mtp_auth/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        url = self._get_url('me')
+        response = self.client.get(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_cannot_access_others_data(self):
+        """
+        Returns 404 when trying to access other's user data.
+        404 because we don't want to leak anything about our
+        db.
```
Comment (SteveMarshall) [inline mtp_api/apps/mtp_auth/tests/test_views.py]: :+1: for 404 here.

File: mtp_api/apps/mtp_auth/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    def setUp(self):
+        super(UserViewTestCase, self).setUp()
+        self.users = make_test_users(users_per_prison=2)
+        self.prisons = Prison.objects.all()
+        make_test_oauth_applications()
+
+    def _get_url(self, username):
+        return reverse('user-detail', kwargs={'username': username})
+
+    def test_401_if_not_logged_in(self):
```
Comment (SteveMarshall) [inline mtp_api/apps/mtp_auth/tests/test_views.py]: Might be more descriptive to call this `test_cannot_access_data_when_not_logged_in`?

**Comment by SteveMarshall** [comment]
Can you tweak the commit message/PR title to be `Add a users endpoint returning my user data`?

File: mtp_api/apps/mtp_auth/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    def test_can_access_my_data(self):
+        user = self.users[0]
+        self.client.force_authenticate(user=user)
+
+        url = self._get_url(user.username)
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.assertEqual(response.data['pk'], user.pk)
```
Comment (SteveMarshall) [inline mtp_api/apps/mtp_auth/tests/test_views.py]: Should there be another test here, to test the user has the correct set of prisons in the serialised data? Something like `test_lists_only_my_prisons_when_logged_in`?

**Comment by marcofucci** [comment]
@r4vi @SteveMarshall OK ?

**Comment by r4vi** [comment]
looks good to me

**Pull request merged**